### ISSUE TITLE
BatchNorm: Documentation: Problem in Inference Equation (and also training)

### ISSUE THREAD

**Issue by MehdiMaboudi** [tensorflow/tensorflow#47608]
in the following links 'sqrt' is missing:
https://www.tensorflow.org/api_docs/python/tf/keras/layers/BatchNormalization#used-in-the-notebooks_1

**During inference:**
instead of  :  (batch - self.moving_mean) / `sqrt`(self.moving_var + epsilon) * gamma + beta
it is written:  (batch - self.moving_mean) / (self.moving_var + epsilon) * gamma + beta.

**During training**
instead of:    (batch - mean(batch)) /`sqrt`(var(batch) + epsilon) * gamma + beta
it is written:  (batch - mean(batch)) / (var(batch) + epsilon) * gamma + beta

**Comment by ymodak** [comment]
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/layers/normalization.py#L57
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/layers/normalization.py#L71
The docs generated on the website don't seem to match with docstrings.

**Comment by lamberta** [comment]
Hmm. I'd expect this to display in the nightly diff but don't see the update: https://www.tensorflow.org/api_docs/python/tf/keras/layers/BatchNormalization?version=nightly

Is the nightly package building?

**Comment by yashk2810** [comment]
I am looking into this. There might be a bug in the diffing algorithm.

**Comment by yashk2810** [comment]
So, I looked through the diffing algorithm and it is working as intended.

The real problem is that tf-nightly doesn't contain the updated docstring.

```
In [11]: sys.version
Out[11]: '3.8.7 (default, Dec 22 2020, 10:37:26) \n[GCC 10.2.0]'

In [12]: tf.__version__
Out[12]: '2.5.0-dev20210309'

In [13]: tf.keras.layers.BatchNormalization.__doc__
Out[13]: 'Layer that normalizes its inputs.\n\n  Batch normalization applies a transformation that maintains the mean output\n  close to 0 and the output standard deviation close to 1.\n\n  Importantly, batch normalization works differently during training and\n  during inference.\n\n  **During training** (i.e. when using `fit()` or when calling the layer/model\n  with the argument `training=True`), the layer normalizes its output using\n  the mean and standard deviation of the current batch of inputs. That is to\n  say, for each channel being normalized, the layer returns\n  `(batch - mean(batch)) / (var(batch) + epsilon) * gamma + beta`, where:\n\n  - `epsilon` is small constant (configurable as part of the constructor\n  arguments)\n  - `gamma` is a learned scaling factor (initialized as 1), which\n  can be disabled by passing `scale=False` to the constructor.\n  - `beta` is a learned offset factor (initialized as 0), which\n  can be disabled by passing `center=False` to the constructor.\n\n  **During inference** (i.e. when using `evaluate()` or `predict()` or when\n  calling the layer/model with the argument `training=False` (which is the\n  default), the layer normalizes its output using a moving average of the\n  mean and standard deviation of the batches it has seen during training. That\n  is to say, it returns\n  `(batch - self.moving_mean) / (self.moving_var + epsilon) * gamma + beta`.\n\n  `self.moving_mean` and `self.moving_var` are non-trainable variables that\n  are updated each time the layer in called in training mode, as such:\n\n  - `moving_mean = moving_mean * momentum + mean(batch) * (1 - momentum)`\n  - `moving_var = moving_var * momentum + var(batch) * (1 - momentum)`\n\n  As such, the layer will only normalize its inputs during inference\n  *after having been trained on data that has similar statistics as the\n  inference data*.\n\n  Args:\n    axis: Integer, the axis that should be normalized (typically the features\n      axis). For instance, after a `Conv2D` layer with\n      `data_format="channels_first"`, set `axis=1` in `BatchNormalization`.\n    momentum: Momentum for the moving average.\n    epsilon: Small float added to variance to avoid dividing by zero.\n    center: If True, add offset of `beta` to normalized tensor. If False, `beta`\n      is ignored.\n    scale: If True, multiply by `gamma`. If False, `gamma` is not used. When the\n      next layer is linear (also e.g. `nn.relu`), this can be disabled since the\n      scaling will be done by the next layer.\n    beta_initializer: Initializer for the beta weight.\n    gamma_initializer: Initializer for the gamma weight.\n    moving_mean_initializer: Initializer for the moving mean.\n    moving_variance_initializer: Initializer for the moving variance.\n    beta_regularizer: Optional regularizer for the beta weight.\n    gamma_regularizer: Optional regularizer for the gamma weight.\n    beta_constraint: Optional constraint for the beta weight.\n    gamma_constraint: Optional constraint for the gamma weight.\n\n  Call arguments:\n    inputs: Input tensor (of any rank).\n    training: Python boolean indicating whether the layer should behave in\n      training mode or in inference mode.\n      - `training=True`: The layer will normalize its inputs using the mean and\n        variance of the current batch of inputs.\n      - `training=False`: The layer will normalize its inputs using the mean and\n        variance of its moving statistics, learned during training.\n\n  Input shape:\n    Arbitrary. Use the keyword argument `input_shape` (tuple of\n    integers, does not include the samples axis) when using this layer as the\n    first layer in a model.\n\n  Output shape:\n    Same shape as input.\n\n  Reference:\n    - [Ioffe and Szegedy, 2015](https://arxiv.org/abs/1502.03167).\n\n  **About setting `layer.trainable = False` on a `BatchNormalization` layer:**\n\n  The meaning of setting `layer.trainable = False` is to freeze the layer,\n  i.e. its internal state will not change during training:\n  its trainable weights will not be updated\n  during `fit()` or `train_on_batch()`, and its state updates will not be run.\n\n  Usually, this does not necessarily mean that the layer is run in inference\n  mode (which is normally controlled by the `training` argument that can\n  be passed when calling a layer). "Frozen state" and "inference mode"\n  are two separate concepts.\n\n  However, in the case of the `BatchNormalization` layer, **setting\n  `trainable = False` on the layer means that the layer will be\n  subsequently run in inference mode** (meaning that it will use\n  the moving mean and the moving variance to normalize the current batch,\n  rather than using the mean and variance of the current batch).\n\n  This behavior has been introduced in TensorFlow 2.0, in order\n  to enable `layer.trainable = False` to produce the most commonly\n  expected behavior in the convnet fine-tuning use case.\n\n  Note that:\n    - Setting `trainable` on an model containing other layers will\n      recursively set the `trainable` value of all inner layers.\n    - If the value of the `trainable`\n      attribute is changed after calling `compile()` on a model,\n      the new value doesn\'t take effect for this model\n      until `compile()` is called again.\n  '
```

I can't see the `sqrt` in the equation here. @mihaimaruseac Do you know what's wrong with the tf-nightly package?

**Comment by yashk2810** [comment]
Ohh, I know what the problem is.

This is the right docstring: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/layers/normalization_v2.py#L205

The one that was updated doesn't contain tf_export decorator. If you update the one that I linked above, then it should show up on the site.

**Comment by lamberta** [comment]
While it's good that the nightly diff is working, I don't think it closes this issue.

Can we update the correct place? Do we need to add a note so contributors of the future update the correct place?

**Comment by lamberta** [comment]
Thanks, Yash.
Looks like this was added in https://github.com/tensorflow/tensorflow/commit/76094af11bffac23375ad7b08f5c7bd6df2e0c7c
@Suraj-Upadhyay Can you please update the correct file? Thanks

**Comment by ymodak** [comment]
The docs are now updated with nightly. Thanks!
https://www.tensorflow.org/api_docs/python/tf/keras/layers/BatchNormalization?version=nightly#used-in-the-notebooks_1
